packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,214 @@ import * as React from 'react';
+      [theme.breakpoints.down('sm')]: {
+        display: 'none',
+      },
+    },
+  }),
+  [`& .${scrollbarSizeClassName}`]: {
     overflowX: 'auto',
     overflowY: 'hidden',
+    // Hide dimensionless scrollbar on MacOS
+    scrollbarWidth: 'none', // Firefox
+    '&::-webkit-scrollbar': {
+      display: 'none', // Safari + Chrome
+    },
   },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `orientation="vertical"`. */
-  scrollableY: {
-    overflowY: 'auto',
-    overflowX: 'hidden',
+}));
+
+const TabsScroller = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Scroller',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styleProps.fixed && styles.fixed),
+          ...(styleProps.hideScrollbar && styles.hideScrollbar),
+          ...(styleProps.scrollableX && styles.scrollableX),
+          ...(styleProps.scrollableY && styles.scrollableY),
+        },
+        styles.scroller || {},
+      );
+    },
   },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `visibleScrollbar={false}`. */
-  hideScrollbar: {
+)(({ styleProps }) => ({
+  position: 'relative',
+  display: 'inline-block',
+  flex: '1 1 auto',
+  whiteSpace: 'nowrap',
+  ...(styleProps.fixed && {
+    overflowX: 'hidden',
+    width: '100%',
+  }),
+  ...(styleProps.hideScrollbar && {
     // Hide dimensionless scrollbar on MacOS
     scrollbarWidth: 'none', // Firefox
     '&::-webkit-scrollbar': {
       display: 'none', // Safari + Chrome
     },
-  },
-  /* Styles applied to the ScrollButtonComponent component. */
-  scrollButtons: {},
-  /* Styles applied to the ScrollButtonComponent component if `allowScrollButtonsMobile={true}`. */
-  scrollButtonsHideMobile: {
-    [theme.breakpoints.down('sm')]: {
-      display: 'none',
+  }),
+  ...(styleProps.scrollableX && {
+    overflowX: 'auto',
+    overflowY: 'hidden',
+  }),
+  ...(styleProps.scrollableY && {
+    overflowY: 'auto',
+    overflowX: 'hidden',
+  }),
+}));
+
+const FlexContainer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'FlexContainer',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styleProps.flexContainerVertical && styles.flexContainerVertical),
+          ...(styleProps.centered && styles.centered),
+        },
+        styles.flexContainer || {},
+      );
     },
   },
-  /* Styles applied to the TabIndicator component. */
-  indicator: {},
-});
+)(({ styleProps }) => ({
+  display: 'flex',
+  ...(styleProps.flexContainerVertical && {
+    flexDirection: 'column',
+  }),
+  ...(styleProps.centered && {
+    justifyContent: 'center',
+  }),
+}));
+
+const TabIndicator = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Indicator',
+    overridesResolver: (props, styles) => styles.indicator,
```

Comment:
```suggestion
```

The `indicator` is a child of the `root`, we should add those overrides in the `Root`'s `overridesResolver`. We need custom `overridesResolver`, only for elements that are not children of the `Root`.

Reply:
by removing `overridesResolver` from TabsIndicator, styles in theme for `indicator` slot still works?
```js
createMuiTheme({
  MuiTabs: {
    styleOverrides: {
      indicator: {
        height: 4,
        borderTopLeftRadius: 4,
        borderTopRightRadius: 4,
      }
    }
  }
})
```
My understanding is in most case the styles that defined in the slot of theme should pass to the slotted component (in this case `indicator` => `TabsIndicator`), is my understanding right?

Reply:
Let's leave it as is, based on the findings in https://github.com/mui-org/material-ui/pull/25850. Originally we were adding in the `Root` overrides resolver all styles using class selectors, something like:

```
  [`& .${tabsClasses.indicator}`]: styles.indicator,
```